### PR TITLE
Add cross compile support

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by isuruf** [comment]
@conda-forge/core this is ready for review

File: recipe/cross_compile_support.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+BUILD_PLATFORM=$(conda info --json | jq -r .platform)
+
+if [ -f .ci_support/${CONFIG}.yaml ]; then
+    HOST_PLATFORM=$(cat .ci_support/${CONFIG}.yaml | shyaml get-value target_platform ${BUILD_PLATFORM})
```
Comment (beckermr) [inline recipe/cross_compile_support.sh:4]: on linux you need FEEDSTOCK_ROOT in the path

File: recipe/download_osx_sdk.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+if [ -f .ci_support/${CONFIG}.yaml ]; then
+   export MACOSX_DEPLOYMENT_TARGET=$(cat .ci_support/${CONFIG}.yaml | shyaml get-value MACOSX_DEPLOYMENT_TARGET.0 10.9)
```
Comment (beckermr) [inline recipe/download_osx_sdk.sh:2]: ditto for this whole file on the FEEDSTOCK_ROOT thing

**Comment by isuruf** [comment]
Thanks for the review. I've used the env variable CI_SUPPORT on osx too.

**Comment by beckermr** [comment]
I think in Linux you need to set the ci support variable too?

**Comment by isuruf** [comment]
It's already set and used elsewhere as well.

**Comment by beckermr** [comment]
Ahhh great. Thanks!

**Comment by beckermr** [comment]
@isuruf I think there is a bug here. 

This python build thinks it is cross compiling:

```
+++ BUILD_PLATFORM=linux-64
+++ '[' -f /home/conda/feedstock_root/.ci_support/linux_python3.7.____cpythontarget_platformlinux-64.yaml ']'
++++ cat /home/conda/feedstock_root/.ci_support/linux_python3.7.____cpythontarget_platformlinux-64.yaml
++++ shyaml get-value target_platform linux-64
+++ HOST_PLATFORM='- linux-64'
+++ HOST_PLATFORM='- linux-64'
+++ [[ - linux-64 != \l\i\n\u\x\-\6\4 ]]
+++ echo 'export CONDA_BUILD_CROSS_COMPILATION=1'
+++ export CONDA_BUILD_CROSS_COMPILATION=1
+++ CONDA_BUILD_CROSS_COMPILATION=1
++ '[' -f /home/conda/feedstock_root/.ci_support/linux_python3.7.____cpythontarget_platformlinux-64.yaml ']'
+++ cat /home/conda/feedstock_root/.ci_support/linux_python3.7.____cpythontarget_platformlinux-64.yaml
+++ shyaml get-value MACOSX_DEPLOYMENT_TARGET.0 0
++ need_osx_sdk=0
++ [[ 0 != \0 ]]
```

link: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=189743&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14

File: recipe/cross_compile_support.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+BUILD_PLATFORM=$(conda info --json | jq -r .platform)
+
+if [ -f ${CI_SUPPORT}/${CONFIG}.yaml ]; then
+    HOST_PLATFORM=$(cat ${CI_SUPPORT}/${CONFIG}.yaml | shyaml get-value target_platform ${BUILD_PLATFORM})
```
Comment (beckermr) [inline recipe/cross_compile_support.sh:4]:
```
    HOST_PLATFORM=$(cat ${CI_SUPPORT}/${CONFIG}.yaml | shyaml get-value target_platform.0 ${BUILD_PLATFORM})
```

Comment (beckermr) [reply recipe/cross_compile_support.sh:4]: I think this is the bug fix.

**Pull request merged**